pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -62,6 +76,48 @@ def visit_call(self, node):
         if isinstance(node.parent, (astroid.For, astroid.Comprehension)):
             self.add_message("consider-iterating-dictionary", node=node)
 
+    def _check_consider_using_maxsplit_arg(self, node: astroid.Call) -> None:
+        """Add message when accessing first or last elements of a str.split() or str.rsplit(),
+        or when split/rsplit with max_split=1 is used"""
+
+        # Check if call is split() or rsplit()
+        if (
+            isinstance(node.func, astroid.Attribute)
+            and node.func.attrname in ("split", "rsplit")
+            and isinstance(utils.safe_infer(node.func), astroid.BoundMethod)
+        ):
+            try:
+                sep = utils.get_argument_from_call(node, 0, "sep").value
+            except utils.NoSuchArgumentError:
+                return
+
+            # Check if maxsplit is set, and ignore checking if maxsplit is > 1
+            try:
+                maxsplit = utils.get_argument_from_call(node, 1, "maxsplit").value
+                if maxsplit > 0:
+                    return
+            except utils.NoSuchArgumentError:
+                maxsplit = 0
+
+            # Check if it's immediately subscripted
+            if isinstance(node.parent, astroid.Subscript):
+                # Check if subscripted with -1/0
+                try:
+                    subscript_value = utils.get_subscript_const_value(node.parent).value
+                except utils.InferredTypeError:
+                    return
+
+                if subscript_value in (-1, 0):
+                    fn_name = node.func.attrname
+                    new_fn = "rsplit" if subscript_value == -1 else "split"
+                    new_name = (
+                        f"{node.as_string().rsplit(fn_name, maxsplit=1)[0]}{new_fn}"
+                        f"(sep='{sep}',maxsplit=1)[{subscript_value}]"
```

Comment:
```suggestion
                        f"('{sep}', maxsplit=1)[{subscript_value}]"
```
The first argument is usually just a positional arg or check if `sep=` is preset in the first place.

Reply:
changed in f60042a2